olive/evaluator/metric_config.py

```diff
@@ -90,3 +90,13 @@ def check_value(cls, v, values):
         if values["type"] in ["min-improvement", "max-degradation"] and v < 0:
             raise ValueError(f"Value must be positive for type {values['type']}")
         return v
+
+
+class MetricResult(ConfigBase):
+    key_for_rank: str
```

Comment:
Can you explain a little bit about `key_for_rank` and `value_for_rank`?

Reply:
Does this mean we only ranked the one subtype? What about the other two? Also, how to cross-rank different metrics with different sub types?